### PR TITLE
Build project with Node 20

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
* Migrate from deprecated vue-service to Vite
* Migrate from Jest to Vitest
* Upgrade to Storybook 7

Notables changes:
* Fixes the memory leak while running unit tests
* Unit tests are a bit faster to run (-5 seconds)
* Hot-reload is available on shops

Backward compatibility kept:
* app.js still exist but has another purpose. Because the way we include files in a template has changed, we load a new file in newer versions, and app.js is kept for old PHP versions to load a light script that will add the new file in the `<head>`

### CODE REVIEW

**Comment by Quetzacoalt91** [comment]
```
dependencies:
+ @googleapis/content 6.2.0
+ @googleapis/oauth2 0.2.0
+ @segment/analytics-next 1.44.0 deprecated
+ @sentry/tracing 7.39.0
- @sentry/vue
+ @sentry/vue 7.39.0
- bootstrap
+ bootstrap 4.6.2
- bootstrap-vue
+ bootstrap-vue 2.22.0
+ chart.js 3.9.1
- chartjs-adapter-date-fns
+ chartjs-adapter-date-fns 2.0.0
- css-loader
+ css-loader 3.6.0
+ date-fns 2.29.2
+ dayjs 1.11.5
+ debounce 1.2.1
+ google-timezones-json 1.0.2
+ lodash.clonedeep 4.5.0
+ lodash.uniqby 4.7.0
+ mktg-with-google-common <- ../../packages/mktg-with-google-common
- prestakit
+ prestakit 1.2.6
+ secure-ls 1.2.6
+ semver 7.3.8
+ showdown 1.9.1
- style-loader
+ style-loader 1.3.0
- ts-loader
+ ts-loader 8.4.0
+ vue 2.7.14
- vue-i18n
+ vue-i18n 8.27.2
- vue-router
+ vue-router 3.6.5
- vue-select
+ vue-select 3.20.0
- vue-showdown
+ vue-showdown 2.4.1
- vuex
+ vuex 3.6.2
- vuex-module-decorators
+ vuex-module-decorators 0.17.0
- vuex-persistedstate
+ vuex-persistedstate 4.1.0 deprecated

devDependencies:
- @storybook/addon-a11y
+ @storybook/addon-a11y 7.0.23
- @storybook/addon-actions
+ @storybook/addon-actions 7.0.23
- @storybook/addon-essentials
+ @storybook/addon-essentials 7.0.23
- @storybook/addon-knobs
+ @storybook/addon-knobs 7.0.2
- @storybook/addon-links
+ @storybook/addon-links 7.0.23
- @storybook/theming
+ @storybook/theming 7.0.23
- @storybook/vue
+ @storybook/vue 7.0.23
- @storybook/vue-vite
+ @storybook/vue-vite 7.0.23
+ @types/debounce 1.2.1
+ @types/lodash.clonedeep 4.5.7
+ @types/showdown 1.9.4
- @typescript-eslint/eslint-plugin
+ @typescript-eslint/eslint-plugin 5.59.9
- @typescript-eslint/parser
+ @typescript-eslint/parser 5.59.9
- @vitejs/plugin-vue
+ @vitejs/plugin-vue 4.2.3
- @vue/eslint-config-airbnb
+ @vue/eslint-config-airbnb 5.3.0
- @vue/eslint-config-typescript
+ @vue/eslint-config-typescript 5.1.0
- @vue/test-utils
+ @vue/test-utils 1.3.0
- chromatic
+ chromatic 5.10.2
+ eslint 8.43.0
- eslint-config-airbnb-base
+ eslint-config-airbnb-base 15.0.0
- eslint-config-prestashop
+ eslint-config-prestashop 0.2.1
- eslint-config-prettier
+ eslint-config-prettier 6.15.0
- eslint-plugin-import
+ eslint-plugin-import 2.26.0
- eslint-plugin-prettier
+ eslint-plugin-prettier 3.4.1
- eslint-plugin-storybook
+ eslint-plugin-storybook 0.6.12
- eslint-plugin-vue
+ eslint-plugin-vue 8.7.1
+ jest-environment-jsdom 29.5.0
+ msw 0.36.8
- msw-storybook-addon
+ msw-storybook-addon 1.8.0
+ react 18.2.0
- react-dom
+ react-dom 18.2.0
+ sass 1.63.6
+ storybook 7.0.23
- ts-jest
+ ts-jest 29.1.0
+ typescript 4.9.5
- vite
+ vite 4.3.9
- vite-plugin-css-injected-by-js
+ vite-plugin-css-injected-by-js 3.1.2
- vite-plugin-react-remove-attributes
+ vite-plugin-react-remove-attributes 1.0.3
- vite-plugin-vue2
+ vite-plugin-vue2 2.0.3
- vitest
+ vitest 0.32.3
- vitest-fetch-mock
+ vitest-fetch-mock 0.2.2
- vue-loader
+ vue-loader 15.10.0
+ vue-template-compiler 2.7.14

 WARN  Issues with peer dependencies found
apps/ui
├─┬ @vitejs/plugin-vue 4.2.3
│ └── ✕ unmet peer vue@^3.2.25: found 2.7.14
├─┬ @vue/eslint-config-airbnb 5.3.0
│ ├── ✕ unmet peer eslint@"^5.16.0 || ^6.1.0 || ^7.2.0": found 8.43.0
│ └─┬ eslint-config-airbnb-base 14.2.1
│   └── ✕ unmet peer eslint@"^5.16.0 || ^6.8.0 || ^7.2.0": found 8.43.0
├─┬ @vue/eslint-config-typescript 5.1.0
│ ├── ✕ unmet peer eslint@"^5.0.0 || ^6.0.0 || ^7.0.0": found 8.43.0
│ ├── ✕ unmet peer @typescript-eslint/eslint-plugin@^2.7.0: found 5.59.9
│ ├── ✕ unmet peer @typescript-eslint/parser@^2.7.0: found 5.59.9
│ └── ✕ unmet peer eslint-plugin-vue@"^5.2.3 || ^6.0.0 || ^7.0.0-0": found 8.7.1
├─┬ vite-plugin-react-remove-attributes 1.0.3
│ └── ✕ unmet peer vite@^2.4.4: found 4.3.9
└─┬ vite-plugin-vue2 2.0.3
  ├─┬ @vue/component-compiler-utils 3.3.0
  │ └─┬ consolidate 0.15.1
  │   ├── ✕ unmet peer react-dom@^16.13.1: found 18.2.0
  │   └── ✕ unmet peer react@^16.13.1: found 18.2.0
  └─┬ consolidate 0.16.0
    ├── ✕ unmet peer react-dom@^16.13.1: found 18.2.0
    └── ✕ unmet peer react@^16.13.1: found 18.2.0

The integrity of 1390 files was checked. This might have caused installation to take longer.
Done in 14.8s
```

File: _dev/apps/ui/.storybook/main.ts [hunk 1]
```diff
@@ -17,21 +17,28 @@
...
+  framework: {
+    name: "@storybook/vue-vite",
+    options: {}
+  },
+  docs: {
+    autodocs: false
+  },
+};
+
+export default config;
```
Comment (fox-john) [inline _dev/apps/ui/.storybook/main.ts:44]: manque un saut de ligne final.

File: _dev/apps/ui/src/components/google-ads-account/google-ads-account-popin-new.spec.ts [hunk 2]
```diff
@@ -1,20 +1,20 @@
 /**
- * @jest-environment jsdom
+ * @vitest-environment jsdom
```
Comment (fox-john) [inline _dev/apps/ui/src/components/google-ads-account/google-ads-account-popin-new.spec.ts:2]: Plutôt que de modifier tout tes headers, tu n'as pas essayé d'ajouter le paramètre dans la config de vitest directement ? Exemple avec le projet Legal : 

image

File: _dev/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "engines": {
-      "node": ">=14",
+      "node": ">=20",
       "pnpm": ">=7"
   }
 }
```
Comment (fox-john) [inline _dev/package.json:6]: Saut de ligne final manquant.

File: views/templates/admin/app.tpl [hunk 4]
```diff
@@ -19,16 +19,20 @@
...
 <script src="https://assets.prestashop3.com/ext/cloudsync-merchant-sync-consent/latest/cloudsync-cdc.js"></script>
 <script src="https://unpkg.com/prestashop_accounts_vue_components@4.3"></script>
 
+{if $psxMktgWithGoogleLiveMode}
+  <script type="module" src="http://localhost:5173/@vite/client"></script>
+  <script type="module" src="http://localhost:5173/src/main.ts"></script>
+{else}
+  <link href="{$pathApp|escape:'htmlall':'UTF-8'}" rel=preload as=script>
+  <script src="{$pathApp|escape:'htmlall':'UTF-8'}" type="module"></script>
+{/if}
```
Comment (fox-john) [inline views/templates/admin/app.tpl:35]: Vous n'avez pas l'app sur CDN ?

Comment (Quetzacoalt91) [reply views/templates/admin/app.tpl:35]: Yes, `$pathApphere` is an absolute path to a local file or a URL.

By default a URL to the CDN will be set here.

File: _dev/apps/ui/src/components/google-ads-account/google-ads-account-popin-new.spec.ts (hunk 2 above)
Comment (Quetzacoalt91) [reply _dev/apps/ui/src/components/google-ads-account/google-ads-account-popin-new.spec.ts:2]: image

Nice

**Review by PoulainMaxime — APPROVED** [review]

**Pull request merged**